Add UnauthorizedException and update UserArgumentResolver

ChernyavckiiPavlo opened the pull request:
**GREEN CITY**

Created UnauthorizedException — a custom runtime exception to explicitly signal unauthenticated access attempts.

Updated UserArgumentResolver to:

Check for missing Principal

Throw UnauthorizedException instead of returning null

Prevent further controller logic from executing with null user

Extended the existing global exception handler to handle UnauthorizedException and return a proper 401 Unauthorized response with a structured body

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Bug Fixes**
  * Improved error handling for unauthorized access by displaying a clear error message and returning a 401 Unauthorized status when authentication is missing.

* **New Features**
  * Added a new error response for unauthorized access, providing users with a specific message when authentication is required.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
Add UnauthorizedException and update UserArgumentResolver to throw on missing principal

Commit:
text formatting

Commit:
text formatting and checkstyle

SashaPog committed:
formatting code

SashaPog committed:
formatting and check for checkstyle code

RomaMocherniuk commented on core/src/main/java/greencity/converters/UserArgumentResolver.java:
```diff
@@ -41,6 +42,11 @@ public boolean supportsParameter(MethodParameter parameter) {
-        return principal != null ? userService.findByEmail(principal.getName()) : null;
+
+        if (principal == null) {
+            throw new UnauthorizedException("Unauthorized access");
```
As you always use the same `Unauthorized access` message it can be moved to the `UnauthorizedException` class

RomaMocherniuk commented on core/src/main/java/greencity/exception/handler/CustomExceptionHandler.java:
```diff
@@ -687,4 +704,26 @@ public final ResponseEntity<Object> handleOpenAIResponseException(OpenAIRequestE
+     *         status
+     */
+    @ExceptionHandler(UnauthorizedException.class)
+    public ResponseEntity<Map<String, String>> handleUnauthorized(UnauthorizedException ex) {
```
Could you please add tests

ChernyavckiiPavlo committed:
Fix test authorization issues and improve UserArgumentResolver integration
- Added .principal(principal) to tests that call controllers from UserArgumentResolver to avoid UnauthorizedException.

ChernyavckiiPavlo committed:
test: cover UserArgumentResolver with unit tests

ChernyavckiiPavlo committed:
test: cover UserArgumentResolver with unit tests

ChernyavckiiPavlo committed:
rename local field in test handleUnauthorizedShouldReturn401
